src/components/ModalFlows/SupportProposal/SupportProposal.js

```diff
@@ -0,0 +1,186 @@
+import React, { useCallback, useMemo, useState } from 'react'
+import BigNumber from '../../../lib/bigNumber'
+import { Button, GU, Info, Slider, textStyle, useTheme } from '@1hive/1hive-ui'
+import useAccountTotalStaked from '../../../hooks/useAccountTotalStaked'
+import { useAppState } from '../../../providers/AppState'
+import { useWallet } from '../../../providers/Wallet'
+
+import { fromDecimals, round, pct } from '../../../utils/math-utils'
+
+import honeySvg from '../../../assets/honey.svg'
+import { formatTokenAmount } from '../../../utils/token-utils'
+
+const SupportProposal = React.memo(function SupportProposal({
+  id,
+  onDone,
+  onStakeToProposal,
+}) {
+  const theme = useTheme()
+
+  const { account } = useWallet()
+  const { accountBalance, stakeToken } = useAppState()
+
+  const totalStaked = useAccountTotalStaked(account)
+  const nonStakedTokens = accountBalance.minus(totalStaked)
+
+  const myStake = useMemo(() => {
+    return totalStaked > 0 ? totalStaked.div(accountBalance).toNumber() : 0
+  }, [totalStaked, accountBalance])
+
+  const [slideValue, amount, handleSliderChange] = useAmount(
+    myStake,
+    stakeToken,
+    accountBalance
+  )
+
+  // Form submit handler
+  const handleSubmit = useCallback(
+    event => {
+      event.preventDefault()
+
+      onStakeToProposal(id, amount.toString(10))
```

Comment:
STEP 3:
so here you need to instead of call the action you need to call the `getTransaction` that you are going to pass from the supportPRoposalScreens and move to the next modal screen, see this example https://github.com/1Hive/honey-pot/blob/3185f4b1c3abfb07cf89b3da4fc1ba64dbb709b6/src/components/ModalFlows/SignAgreementScreens/SignOverview.js#L16-L23